app/Notifications/NewRequestBounty.php

```diff
@@ -42,6 +43,32 @@ public function via(object $notifiable): array
         return ['database'];
     }
 
+    /**
+     * Determine if the notification should be sent.
+     *
+     * @return bool
+     */
+    public function shouldSend(User $notifiable): bool
+    {
+        $targetGroup = 'json_request_groups';
+
+        if ($notifiable->notification?->block_notifications == 1) {
+            return false;
+        }
+
+        if (!$notifiable->notification?->show_request_bounty) {
+            return false;
+        }
+
+        if (\is_array($notifiable->notification->$targetGroup)) {
+            // If the sender's group ID is found in the "Block all notifications from the selected groups" array,
+            // the expression will return false.
+            return !\in_array($this->bounty->user->group->id, $notifiable->notification->$targetGroup, true);
+        }
+
+        return true;
```

Comment:
We should be able to assume that it will always be an array (We made migrations to ensure this a few years ago). Also, it's better to directly access the UserNotification property `json_request_groups`. Also `->group->id` -> `->group_id`. Also remove `return @bool` since it's already indicated in the function signature. This goes for all the other notifications too.

```suggestion
     */
    public function shouldSend(User $notifiable): bool
    {
        if ($notifiable->notification?->block_notifications == 1) {
            return false;
        }

        if (!$notifiable->notification?->show_request_bounty) {
            return false;
        }

        // If the sender's group ID is found in the "Block all notifications from the selected groups" array,
        // the expression will return false.
        return !\in_array($this->bounty->user->group_id, $notifiable->notification->json_request_groups, true);
```